File: #0458.poor-pigs.cpp
class Solution {
public:
    int poorPigs(int buckets, int minutesToDie, int minutesToTest) {

        int base = minutesToTest / minutesToDie + 1;
        int pigs = 0;
        while (std::pow(base, pigs) < buckets) {
            pigs++;
        }
        return pigs;
    }
};

File: #0458.poor-pigs.js
/**
 * @param {number} buckets
 * @param {number} minutesToDie
 * @param {number} minutesToTest
 * @return {number}
 */
var poorPigs = function(buckets, minutesToDie, minutesToTest) {
    let max = minutesToTest / minutesToDie + 1;
    let result = 0;
    while (Math.pow(max, result) < buckets) {
        ++result;
    }
    return result;
};
